packages/plugin-response-cache/src/__tests__/integration.test.ts

```diff
@@ -0,0 +1,505 @@
+import {
+  ApolloServer,
+  GraphQLRequest,
+  GraphQLRequestContext,
+} from '@apollo/server';
+import { startStandaloneServer } from '@apollo/server/standalone';
+import ApolloServerPluginResponseCache from '../index.js';
+import request, { type Response } from 'supertest';
+
+describe('Response caching', () => {
+  beforeAll(() => {
+    // These tests use the `FakeableTTLTestingCache` whose maxAge feature is
```

Comment:
If this is literally only mocking out the `Date.now()` call in the test-specific FakeableTTLTestingCache class, then I think we can just have the FakeableTTLTestingCache have a `Date` property and a method to increment it, and use that instead of `Date.now()`? Or is there some non-test code that this is mocking out too?  (A comment noting that we've had trouble mocking timers with InMemoryLRUCache and that's why we use FakeableTTLTestingCache seems reasonable though.)

Reply:
Updated comments to paint a better picture. `Date` needs to be mocked because it's an internal detail of the plugin itself (used for calculating the `age` header as well as `cacheTime` when a value is set).

Having the mock cache use `Date` as well lets us piggyback on the fact that `Date` is already mocked, so we can mock exactly one thing and also only call `advanceTimersByTime` on one thing (Jest) and not also have to manage the cache's internal time separately.